### ISSUE TITLE
Gunicorn LWRP

### ISSUE THREAD

**Issue by tschuy** [osuosl-cookbooks/python-webapp#18]
We need to set up Gunicorn to fully replace the application_python cookbook.

This will:

1) set up supervisor running at boot
2) set up a gunicorn instance in supervisor, automatically
3) se tup gunicorn to talk to the python_webapp on the right port (user set)

What I'm thinking of:

```
gunicorn do
  port node['whats_fresh']['gunicorn_port']
  loglevel 'debug'
end
```

This would need to go inside the python_webapp LWRP so it knows where the template should point it.

**Comment by tschuy** [comment]
Currently, from a converged What's Fresh instance:

```
[root@seagrant-staging ~]# cat /etc/supervisor.d/whats_fresh.conf 
[program:whats_fresh]
command=/opt/whats_fresh/shared/env/bin/gunicorn whats_fresh.wsgi:application -c /opt/whats_fresh/shared/gunicorn_config.py
process_name=%(program_name)s
numprocs=1
numprocs_start=0
priority=999
autostart=true
autorestart=true
startsecs=1
startretries=3
exitcodes=0,2
stopsignal=TERM
stopwaitsecs=10
user=root
redirect_stderr=false
stdout_logfile=AUTO
stdout_logfile_maxbytes=50MB
stdout_logfile_backups=10
stdout_capture_maxbytes=0
stdout_events_enabled=false
stderr_logfile=AUTO
stderr_logfile_maxbytes=50MB
stderr_logfile_backups=10
stderr_capture_maxbytes=0
stderr_events_enabled=false
directory=/opt/whats_fresh/current/
serverurl=AUTO
```

and

```
[root@seagrant-staging ~]# cat /opt/whats_fresh/shared/gunicorn_config.py 
##
# Gunicorn config at 
# Managed by Chef - Local Changes will be Nuked from Orbit (just to be sure)
##

# What ports/sockets to listen on, and what options for them.
bind = "0.0.0.0:8080"

# The maximum number of pending connections
backlog = 2048

# What the timeout for killing busy workers is, in seconds
timeout = 30

# How long to wait for requests on a Keep-Alive connection, in seconds
keepalive = 2

# The maxium number of requests a worker will process before restarting
max_requests = 0

# Whether the app should be pre-loaded
preload_app = False

# How many worker processes
workers = 8

# Type of worker to use
worker_class = "sync"
```

**Comment by tschuy** [comment]
I've gone through the gunicorn and supervisor cookbooks and this won't be hard to do. Rather than create a multi-layer LWRP, which will increase the complexity of the LWRP, I'm just going to make something like a `gunicorn` bool and a  `gunicorn_port` integer.

**Comment by tschuy** [comment]
We can get rid of the gunicorn bool. If you want to use gunicorn, give it a port.

**Comment by tschuy** [comment]
Fixed by #20
